core/src/cluster_slots.rs

```diff
@@ -106,28 +106,30 @@ impl ClusterSlots {
         }
     }
 
-    pub fn compute_weights(&self, slot: Slot, repair_peers: &[ContactInfo]) -> Vec<(u64, usize)> {
-        let slot_peers = self.lookup(slot);
+    pub fn compute_weights(&self, slot: Slot, repair_peers: &[ContactInfo]) -> Vec<u64> {
+        let stakes = {
+            let validator_stakes = self.validator_stakes.read().unwrap();
+            repair_peers
+                .iter()
+                .map(|peer| {
+                    validator_stakes
+                        .get(&peer.id)
+                        .map(|node| node.total_stake)
+                        .unwrap_or(0)
+                        + 1
+                })
+                .collect()
+        };
+        let slot_peers = match self.lookup(slot) {
```

Comment:
heh unrelated to this PR, but I'm realizing that this `lookup` is probably not that useful, especially since we cache the value the first time time we compute weights for this slot. Because repair usually runs on missing shreds in the middle of a slot, not many people will have the slot completed, so the `lookup()` here will return a `slot_peers` that's mostly empty.